tests/cupy_tests/sorting_tests/test_sort.py

```diff
@@ -416,6 +416,35 @@ def test_msort_multi_dim(self, xp, dtype):
         return xp.msort(a)
 
 
+@testing.gpu
+class TestSort_complex(unittest.TestCase):
+
+    def test_sort_complex_zero_dim(self):
+        for xp in (numpy, cupy):
+            a = testing.shaped_random((), xp)
+            with pytest.raises(numpy.AxisError):
+                xp.sort_complex(a)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_list_equal()
+    def test_sort_complex_1dim(self, xp, dtype):
+        a = testing.shaped_random((100,), xp, dtype)
+        return xp.sort_complex(a)
```

Comment:
https://github.com/cupy/cupy/pull/3348/files#r429978698 
```suggestion
        return [a, xp.sort_complex(a)]
```

Reply:
```
 if a.dtype.char == 'e':
     a = a.astype('D') 
```
These 2 lines were intended to support fp 16 (or I can suppress it from ``` testing.for_dtypes() ```) The suggested change ( return [a, xp.sort_complex(a)]) fails with all tests whether I keep or remove these 2 lines.

Reply:
So please fix your implementation to pass this check.

Reply:
Hi @Dahlia-Chehata I've enabled `float16` sort early on, so if NumPy requires the output dtype to be `float64`, you should sort first and then cast for better performance.